APMrover2/mode.h

```diff
@@ -481,3 +485,27 @@ class ModeFollow : public Mode
 
     bool _enter() override;
 };
+
+class ModeSimple : public Mode
+{
+public:
+
+    uint32_t mode_number() const override { return SIMPLE; }
+    const char *name4() const override { return "SMPL"; }
+
+    // methods that affect movement of the vehicle in this mode
+    void update() override;
+    void init_heading();
+
+private:
+
+    // simple type enum used for SIMPLE_TYPE parameter
+    enum simple_type {
+        Simple_InitialHeading = 0,
+        Simple_CardinalDirections = 1,
+    };
+
+    float _initial_heading_cd;  // vehicle heading (in centi-degrees) at moment vehicle was armed
```

Comment:
floats and centidegrees make me sad.  I blame the ahrs.

Reply:
.. there's a lot more centi-degrees where that came from.. :-).  I agree we should fix this though (in a separate PR) and it's not too hard in Rover..